git_info/git_summarizer.py

```diff
@@ -0,0 +1,84 @@
+import json
+import logging
+import traceback
+from typing import Tuple
+
+from dulwich import porcelain
+from dulwich.objects import ShaFile
+from dulwich.repo import Repo
+import difflib
+
+
+def get_diffs(repo: Repo, old_sha: ShaFile, new_sha: ShaFile) -> Tuple[int, int]:
+    """
+    Gets information about diffs in particular change
+    :param repo:        repository path
+    :param old_sha:
+    :param new_sha:
+    :return: tuple with number of added strings as first value and deleted as second
+    """
+    diffs = difflib.unified_diff(
+        repo.get_object(old_sha).data.decode().splitlines(),
+        repo.get_object(new_sha).data.decode().splitlines()
+    )
+    added, deleted = 0, 0
+    for diff in diffs:
+        match diff[:1]:
+            case ['+', char] if char != '+':
+                added += 1
+            case ['-', char] if char != '-':
+                deleted += 1
+    return added, deleted
+
+
+def save_commits_data(repo_dir: str, data_dir: str) -> None:
+    """
+    Saves data about commits in provided directory
+    :param repo_dir:    repository path
+    :param data_dir:    directory to save data
+    :return:
```

Comment:
`return` could be deleted because you show with typing that `None` will be returned